engine/src/main/java/org/hibernate/search/engine/mapper/schema/management/SearchSchemaCollector.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.mapper.schema.management;
```

Comment:
Not sure this is the right place... Either you consider this is about the backend/index and then it should be in `engine.backend.*`, or you consider it's about the mapper and then it should be in the `pojo-base` module so that we can add methods specific to the POJO mappers.

What about the [original plan](https://github.com/hibernate/hibernate-search/pull/3437#discussion_r1130658256) of having `IndexSchemaCollector` live in the engine in a `backend.schema.management` package, and `SearchSchemaCollector` live in `pojo-base` (and extending `IndexSchemaCollector`)?